src/common/dns_utils.cpp

```diff
@@ -327,10 +299,14 @@ std::vector<std::string> DNSResolver::get_record(const std::string& url, int rec
   std::vector<std::string> addresses;
   dnssec_available = false;
   dnssec_valid = false;
-
-  // destructor takes care of cleanup
-  ub_result_ptr result;
-
+  
+  ub_result *result;
+  // Make sure we are cleaning after result.
+  epee::misc_utils::auto_scope_leave_caller scope_exit_handler =
```

Comment:
Why not combine these into a `std::unique_ptr` with a custom deleter?

Reply:
No easy way of handling `**` argument via `unique_ptr` [1]. The `result` is used as parameter [2].


1. https://stackoverflow.com/a/24543176
2. https://github.com/monero-project/monero/pull/9090/files/430b102aef0151820a27eafe232afa40ba8e3d8f#diff-ed01ce7362fc3c2d5c7adc0a0be9742bbc6c7558825914f6a2d288b1e0f552fcR313